src/FileSystem/FileSystemHelper.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace OomphInc\WASP\FileSystem;
+
+use RuntimeException;
+
+abstract class FileSystemHelper {
+
+	/**
+	 * Recursively flatten a nested files array.
+	 * @param  array  $files paths to files or an array in the form ['subdir' => [files]]
+	 * @param  string [$dir]   starting dir
+	 * @return array         flatten array with resolved file names
+	 */
+	public static function flattenFileArray($files, $dir = null) {
+		$out = [];
+		$dir = $dir ? rtrim($dir, '/') . '/' : '';
+		foreach ($files as $file) {
+			if (is_string($file)) {
+				$out[] = $dir ? $dir . $file : $file;
```

Comment:
if you default `$dir = ''` instead of `$dir = null` this can just become
```php
$out[] = $dir . $file

Reply:
I feel like there was a reason I did it this way, lemme review

Reply:
yeah I don't see any reason why that wouldn't work

Reply:
the default value doesn't have to be `''` for this to work since it becomes that on L#17 anyway

Reply:
but maybe it's more logical that way